### DISCUSSION TITLE
How to pass data in redirects?

### DISCUSSION THREAD

**Discussion by pquerner** [pleerock/routing-controllers#1328] [Q&A]
Hello!

I have a route which update some model, but in order to do that it needs to run some checks.
If the checks fail, I want to redirect back to the "edit" page and add a message to the frontend, what went wrong.

I thought I could do it like that:

```
@Controller('/pn')
export class PushNotificationsController {
    @Post('/update/:id([0-9]+)')
    async updatePn(@Req() request: Request, @Res() response: Response, @Param('id') id: number, @Body() pn: PushNotificationCreationAttributes) {
        response.locals.foo = "baz"; // set something on the response "locals" object
        response.redirect('/pn/edit/' + id);
        return response;

@Get('/edit/:id([0-9]+)')
    @Render('page/pn/edit.njk')
    async edit(@Res() response: Response, @Req() request: Request, @Param('id') id: number) {
// response.locals is empty {} (object)
//...
}
```

Setting on the express instance itself (app.set) is persistant for all following request. (Which I do not want)

**Comment by pquerner** [comment]
I use sessions for that now.

=> https://github.com/tj/connect-redis

**Comment by attilaorosz** [comment]
@pquerner Indeed session seems like the best bet for this.
